src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts

```diff
@@ -5,11 +5,22 @@
 
 import { OnDestroy } from '@angular/core';
 import { AngularDisposable } from 'sql/base/browser/lifecycle';
+import { ICellEditorProvider } from 'sql/workbench/services/notebook/browser/notebookService';
+import { BaseTextEditor } from 'vs/workbench/browser/parts/editor/textEditor';
+import { NotebookRange } from 'sql/workbench/contrib/notebook/browser/models/modelInterfaces';
 
-export abstract class CellView extends AngularDisposable implements OnDestroy {
+export abstract class CellView extends AngularDisposable implements OnDestroy, ICellEditorProvider {
 	constructor() {
 		super();
 	}
 
 	public abstract layout(): void;
+
+	public abstract getEditor(): BaseTextEditor;
+
+	public abstract hasEditor(): boolean;
+
+	public abstract cellGuid(): string;
```

Comment:
I'd suggest having this and getEditor just be readonly properties instead. No reason to have them as functions as far as I can tell.